include/exadg/solvers_and_preconditioners/multigrid/coarse_grid_solvers.h

```diff
@@ -166,82 +166,89 @@ class MGCoarseKrylov
     if(additional_data.operator_is_singular)
       set_zero_mean_value(r);
 
-    if(additional_data.preconditioner == MultigridCoarseGridPreconditioner::TrilinosAMG)
+    if(additional_data.preconditioner == MultigridCoarseGridPreconditioner::AMG)
     {
-      // create temporal vectors of type NumberAMG (double)
-      VectorTypeAMG dst_tri;
-      dst_tri.reinit(dst, false);
-      VectorTypeAMG src_tri;
-      src_tri.reinit(r, true);
-      src_tri.copy_locally_owned_data_from(r);
-
-      std::shared_ptr<PreconditionerTrilinosAMG<Operator, NumberAMG>> coarse_operator =
-        std::dynamic_pointer_cast<PreconditionerTrilinosAMG<Operator, NumberAMG>>(
-          preconditioner_amg);
+      if(additional_data.amg_data.amg_type == AMGType::Trilinos)
+      {
+        // create temporal vectors of type NumberAMG (double)
+        VectorTypeAMG dst_tri;
+        dst_tri.reinit(dst, false);
+        VectorTypeAMG src_tri;
+        src_tri.reinit(r, true);
+        src_tri.copy_locally_owned_data_from(r);
+
+        std::shared_ptr<PreconditionerTrilinosAMG<Operator, NumberAMG>> coarse_operator =
```

Comment:
For consistency, we would then maybe also rename this class into `PreconditionerML`?

The name `PreconditionerBoomerAMG` used below appears to be fine already.

Reply:
Done in commit [d0a2ff9](https://github.com/exadg/exadg/pull/58/commits/d0a2ff93d87cd124a4426a80462886196481f72a).